examples/mapreduce/mapreduce.cc

```diff
@@ -0,0 +1,270 @@
+#include <fstream>
+#include <algorithm>
+#include <iostream>
+#include <stdlib.h> // std::atoi, std::rand()
+#include <iomanip>
+#include <string>
+#include <memory>
+#include <map>
+#include <chrono>
+#include <filesystem>
+#include "ttg.h"
+
+#define BLOCK_SIZE 32
+
+using namespace ttg;
+
+template<typename T>
+using Key = std::pair<std::pair<std::string, T>, T>;

```

Comment:
Do we need to send the filename as part of the key? Is it used after reading from the file?

Reply:
This example counts words from multiple files. In order to keep the keys unique, I included filename in the key. We can probably come up with a different way of having unique keys as well without using the filename like a file ID for example.